android/src/main/java/ru/vvdev/yamap/RNYamapModule.java

```diff
@@ -45,14 +46,30 @@ public String getName() {
     }
 
     @ReactMethod
-    public void init(final String apiKey) {
+    public void init(final String apiKey, final Promise promise) {
         runOnUiThread(new Thread(new Runnable() {
             @Override
             public void run() {
-                MapKitFactory.setApiKey(apiKey);
-                MapKitFactory.initialize(reactContext);
-                TransportFactory.initialize(reactContext);
-                MapKitFactory.getInstance().onStart();
+                Error apiKeyError = null;
+                try {
+                    // In case when android application reloads during development
+                    // MapKitFactory is already initialized
+                    // And setting api key leads to crash
+                    try {
+                        MapKitFactory.setApiKey(apiKey);
+                    } catch (Error error) {
+                        apiKeyError = error;
+                    }
+                    MapKitFactory.initialize(reactContext);
+                    TransportFactory.initialize(reactContext);
+                    MapKitFactory.getInstance().onStart();
+                    promise.resolve(null);
+                } catch (Error error) {
+                    if(apiKeyError != null) {
+                        promise.reject(apiKeyError);
+                    }
+                    promise.reject(error);
+                }
```

Comment:
Предложил бы все обернуть в try/catch
```suggestion
                    // In case when android application reloads during development
                    // MapKitFactory is already initialized
                    // And setting api key leads to crash
                try {
                    MapKitFactory.setApiKey(apiKey);
                    MapKitFactory.initialize(reactContext);
                    TransportFactory.initialize(reactContext);
                    MapKitFactory.getInstance().onStart();
                    promise.resolve(null);
                } catch (Error error) {
                    promise.reject("YMAPKIT_ERROR ", "Error with initiating YMapKit: "+ error.getMessage());
                }
```